pkg/catalog/retry.go

```diff
@@ -0,0 +1,74 @@
+package catalog
+
+import (
+	"github.com/openservicemesh/osm/pkg/apis/policy/v1alpha1"
+	"github.com/openservicemesh/osm/pkg/identity"
+	"github.com/openservicemesh/osm/pkg/policy"
+	"github.com/openservicemesh/osm/pkg/service"
+)
+
+// GetRetryPolicy returns the RetryPolicySpec for the given downstream identity and upstream service
+// If a source and destination has more than one retry policy that can be applied, the more
+// specific retry policy will be used. Types of retry policy:
+// - specific retry policy that specifies the source and destination
+// - wildcard destination service retry policy
+// - wildcard destination service and namespace retry policy
+func (mc *MeshCatalog) GetRetryPolicy(downstreamIdentity identity.ServiceIdentity, upstreamSvc service.MeshService) *v1alpha1.RetryPolicySpec {
+	if !mc.configurator.GetFeatureFlags().EnableRetryPolicy {
+		log.Trace().Msgf("Retry policy flag not enabled")
+		return nil
+	}
+	src := downstreamIdentity.ToK8sServiceAccount()
+
+	// List the retry policies for the source
+	retryPolicies := mc.policyController.ListRetryPolicies(src)
+	if retryPolicies == nil {
+		log.Trace().Msgf("Did not find retry policy for downstream service %s", src)
+		return nil
+	}
+	var retry *v1alpha1.RetryPolicySpec
+	prioritizeRetry := false
```

Comment:
What does `prioritizeRetry` mean? Kindly clarify with a comment above the variable.

Reply:
Removed it as it's for supporting wildcard destinations, but prioritizeRetry would be true when a more specific retry policy is found